ee/server/service/certificate_authorities.go

```diff
@@ -372,3 +380,209 @@ func (svc *Service) DeleteCertificateAuthority(ctx context.Context, certificateA
+	if digicert.Name != nil {
+		if err := validateCAName(*digicert.Name, errPrefix); err != nil {
+			return err
+		}
+	}
+	if digicert.URL != nil {
+		if err := validateURL(*digicert.URL, "DigiCert", errPrefix); err != nil {
+			return err
+		}
+	}
+	if digicert.APIToken != nil && *digicert.APIToken == "" {
+		return &fleet.BadRequestError{
+			Message: fmt.Sprintf("%sInvalid DigiCert API token. Please correct and try again.", errPrefix),
+		}
+	}
+	if digicert.ProfileID != nil && *digicert.ProfileID == "" {
+		return &fleet.BadRequestError{
+			Message: fmt.Sprintf("%sInvalid profile GUID. Please correct and try again.", errPrefix),
+		}
+	}
+	if digicert.CertificateCommonName != nil {
+		if err := validateDigicertCACN(*digicert.CertificateCommonName, errPrefix); err != nil {
+			return err
+		}
+	}
+	if digicert.CertificateUserPrincipalNames != nil {
+		if err := validateDigicertUserPrincipalNames(*digicert.CertificateUserPrincipalNames, errPrefix); err != nil {
+			return err
+		}
+	}
+	if digicert.CertificateSeatID != nil {
+		if err := validateDigicertSeatID(*digicert.CertificateSeatID, errPrefix); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (svc *Service) validateHydrantUpdate(hydrant *fleet.HydrantCAUpdatePayload, errPrefix string) error {
+	if hydrant.Name != nil {
+		if err := validateCAName(*hydrant.Name, errPrefix); err != nil {
+			return err
+		}
+	}
+	if hydrant.URL != nil {
+		if err := validateURL(*hydrant.URL, "Hydrant", errPrefix); err != nil {
+			return err
+		}
+	}
+	if hydrant.ClientID != nil && *hydrant.ClientID == "" {
+		return &fleet.BadRequestError{
+			Message: fmt.Sprintf("%sInvalid Hydrant Client ID. Please correct and try again.", errPrefix),
+		}
+	}
+	if hydrant.ClientSecret != nil && *hydrant.ClientSecret == "" {
+		return &fleet.BadRequestError{
+			Message: fmt.Sprintf("%sInvalid Hydrant Client Secret. Please correct and try again.", errPrefix),
+		}
+	}
+	return nil
+}
+
+func (svc *Service) validateNDESSCEPProxyUpdate(ctx context.Context, ndesSCEP *fleet.NDESSCEPProxyCAUpdatePayload, errPrefix string) error {
+	// some methods in this fuction require the NDESSCEPProxyCA type so we convert the ndes update payload here
+	NDESProxy := fleet.NDESSCEPProxyCA{
+		URL:      *ndesSCEP.URL,
+		AdminURL: *ndesSCEP.AdminURL,
+		Username: *ndesSCEP.Username,
+		Password: *ndesSCEP.Password,
+	}
+
+	if ndesSCEP.URL != nil {
+		if err := validateURL(*ndesSCEP.URL, "NDES SCEP", errPrefix); err != nil {
+			return err
+		}
+	}
+	if err := svc.scepConfigService.ValidateSCEPURL(ctx, *ndesSCEP.URL); err != nil {
+		level.Error(svc.logger).Log("msg", "Failed to validate NDES SCEP URL", "err", err)
+		return &fleet.BadRequestError{Message: fmt.Sprintf("%sInvalid SCEP URL. Please correct and try again.", errPrefix)}
+	}
+	if err := svc.scepConfigService.ValidateNDESSCEPAdminURL(ctx, NDESProxy); err != nil {
+		level.Error(svc.logger).Log("msg", "Failed to validate NDES SCEP admin URL", "err", err)
```

Comment:
I could image these suffer from the same issue:

If we do not alter the URL it will be empty in the update payload, and this will panic/fail. We need to guard it with a URL != nil check.

There is the above != nil, but that is just for validating it's a valid request URI, and then if it is nil it will still be called.